File: LinkedLists.py
class Node:
    def __init__(self, data):
        self.data = data
        self.value = ""
        self.left = None
        self.right = None
        self.up = None
        self.down = None

class TernaryLinkedList:
    def __init__(self):
        self.head = None
        self.end = None
        
    def print_list(self):
        temp = self.head
        while temp:
            curr = temp
            while curr:
                print(curr.data, end=" -> ")
                curr = curr.right
            print("None")
            temp = temp.down
        print("None")

    def get_node(self, row_number, col_number):
        temp = self.head
        col = 0
        while col < col_number:
            temp = temp.right
            col += 1
        row = 0
        while row < row_number:
            temp = temp.down
            row += 1
        return temp

    def insert_node(self, i, j):
        node = Node(f"{i}, {j}")
        if i == 0 and j == 0:
            self.head = node
            self.end = node
            return

        if j > 0:
            node.left = self.end
            self.end.right = node
        elif j == 0 and i > 0:
            prev_row_start = self.get_node(i - 1, 0)
            prev_row_start.down = node
            node.up = prev_row_start

        if i > 0:
            upper_node = self.get_node(i - 1, j)
            node.up = upper_node
            upper_node.down = node

        self.end = node

File: LinkedLists.js
class Node {
    constructor(row, col) {
        this.ref = [row, col]
        this.value = ""
        this.formula = "";
        this.left = null
        this.right = null
        this.up = null
        this.down = null
    }
}

class TernaryLinkedList {
    constructor() {
        this.head = null
        this.end = null
    }

    getNode(rowNumber, colNumber) {
        let temp = this.head
        let col = 0
        while (col++ < colNumber) temp = temp.right
        let row = 0
        while (row++ < rowNumber) temp = temp.down
        return temp
    }

    columnToIndex(colLetters) {
        let colIndex = 0;
        for (let i = 0; i < colLetters.length; i++) {
            colIndex = colIndex * 26 + (colLetters.charCodeAt(i) - "A".charCodeAt(0) + 1);
        }
        return colIndex - 1;
    }

    insertNode(i, j) {
        const node = new Node(i,j);
        if (i === 0 && j === 0) {
            this.head = node;
            this.end = node;
            return;
        }
        if (j > 0) {
            node.left = this.end; 
            this.end.right = node;
        } else if (j === 0 && i > 0) {
            let prevRowStart = this.getNode(i - 1, 0);
            prevRowStart.down = node;
            node.up = prevRowStart;
        }
        if (i > 0) {
            let upperNode = this.getNode(i - 1, j);
            node.up = upperNode;
            upperNode.down = node;
        }
        this.end = node;
    }
}